components/mediation-registry/org.wso2.carbon.mediation.registry/src/main/java/org/wso2/carbon/mediation/registry/MicroIntegratorRegistry.java

```diff
@@ -492,8 +491,13 @@ public void newNonEmptyResource(String path, boolean isDirectory, String content
             String parent = getParentPath(targetPath, isDirectory);
             String fileName = getResourceName(targetPath);
 
+            Properties metadata = new Properties();
+            if (mediaType != null) {
+                metadata.setProperty(METADATA_KEY_MEDIA_TYPE, mediaType);
+            }
+
             try {
-                writeToFile(new URI(parent), fileName, content);
+                writeToFile(new URI(parent), fileName, content, metadata);
             } catch (Exception e) {
```

Comment:
Shall we catch the specific exception here, rather generic `Exception`? 
Is there a reason for handling the exception here and log, rather propergating it to the caller?

Reply:
`handleException` method which is called inside the catch clause throws an `SynapseException`.